server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -826,7 +827,14 @@ public final void writeOptionalInstant(@Nullable Instant instant) throws IOExcep
                             o.writeByte((byte) 25);
                         }
                         o.writeCollection((Set<?>) v, StreamOutput::writeGenericValue);
-                    }
+                    }),
+            entry(
+                // TODO: improve serialization of BigInteger
+                BigInteger.class,
+                (o, v) -> {
+                    o.writeByte((byte) 26);
+                    o.writeString(v.toString());
+                }
```

Comment:
Can you fix the indentation of the blocks ?

Reply:
Addressed in 9e057c0